lib/WebpackOptionsApply.js

```diff
@@ -236,9 +237,21 @@ class WebpackOptionsApply extends OptionsApply {
 			(options.output.enabledLibraryTypes);
 
 		if (enabledLibraryTypes.length > 0) {
+			let once = true;
 			for (const type of enabledLibraryTypes) {
 				const EnableLibraryPlugin = require("./library/EnableLibraryPlugin");
-				new EnableLibraryPlugin(type).apply(compiler);
+				new EnableLibraryPlugin(type, {
+					// eslint-disable-next-line no-loop-func
+					additionalApply: () => {
+						if (!once) return;
+						once = false;
+						// We rely on `exportInfo` to generate the `export statement` in certain library bundles.
+						// Therefore, we ignore the disabling of `optimization.providedExport` and continue to apply `FlagDependencyExportsPlugin`.
+						if (!options.optimization.providedExports) {
+							new FlagDependencyExportsPlugin().apply(compiler);
+						}
+					}
+				}).apply(compiler);
```

Comment:
Looks good, what is about apply it only for require libraries, i.e. where we use `getExportsInfo()`/`getExportInfo()`?

Reply:
Don't worry, thank you for your help